nocopy_linkbuffer.go

```diff
@@ -124,28 +129,50 @@ func (b *UnsafeLinkBuffer) Peek(n int) (p []byte, err error) {
 	if b.isSingleNode(n) {
 		return b.read.Peek(n), nil
 	}
+
 	// multiple nodes
-	var pIdx int
-	if block1k < n && n <= mallocMax {
-		p = malloc(n, n)
-		b.caches = append(b.caches, p)
-	} else {
-		p = dirtmake.Bytes(n, n)
-	}
-	var node = b.read
-	var l int
-	for ack := n; ack > 0; ack = ack - l {
-		l = node.Len()
-		if l >= ack {
-			pIdx += copy(p[pIdx:], node.Peek(ack))
-			break
-		} else if l > 0 {
-			pIdx += copy(p[pIdx:], node.Peek(l))
+	// always use malloc, since we will reuse b.cachePeek
+	if b.cachePeek != nil && cap(b.cachePeek) < n {
```

Comment:
here has a potential risk, because currently `malloc(4095, 4095)` may return a slice (len=4095, 4096) ...... 

but I fixed it in recent commit: https://github.com/bytedance/gopkg/pull/214/commits/65caa4f9bb95db9488ce7356775c36ab86b6ad19 , need upgrade gopkg after it merged